internal/controller/siteconfig_validator.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	sc_v1alpha1 "github.com/sakhoury/siteconfig/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func isValidJsonString(input string) bool {
+	var result interface{}
+	err := json.Unmarshal([]byte(input), &result)
+	return err == nil
+}
+
+// ValidateSiteConfig validates the given SiteConfig, returns an error if validation fails, returns nil if it succeeds
+func ValidateSiteConfig(ctx context.Context, c client.Client, siteConfig *sc_v1alpha1.SiteConfig) error {
```

Comment:
This doesn't need to be caps if it's not being used from outside the package.

Reply:
Changed